### PR TITLE
Alerting: replace usage of simplejson to json.RawMessage in NotificationChannelConfig

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This is part of efforts to extract alerting code to a separate repository. 
It removes the usage of `simpleJson` from the NotificationChannelConfig struct. There are a few notifiers that have not been updated to use the default json marshaller, I just updated them to use simplejson on its own.

Also, PR introduces a type RawMessage that is an alias for json.RawMessage. The reason for alias is that some models support YAML deserialization and the field type is not supported that. Although, we do not seem to use it, there are tests that check unmarshaling from YAML. Therefore, I chose to implement simple marshaling for RawMessage to\from YAML rather than deleting the code.

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/util.go [hunk 1]
```diff
@@ -170,20 +168,12 @@ type NotificationChannelConfig struct {
 	Name                  string            `json:"name"`
 	Type                  string            `json:"type"`
 	DisableResolveMessage bool              `json:"disableResolveMessage"`
-	Settings              *simplejson.Json  `json:"settings"`
+	Settings              json.RawMessage   `json:"settings"`
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/notifier/channels/util.go:171]: this is the essence of this PR

**Review by gotjosh — APPROVED** [review]
LGTM

**Comment by armandgrillet** [comment]
```
--- FAIL: Test_PostableApiReceiver_Unmarshaling_YAML (0.00s)
    --- FAIL: Test_PostableApiReceiver_Unmarshaling_YAML/grafana_receivers (0.00s)
        alertmanager_test.go:403: 
            	Error Trace:	/drone/src/pkg/services/ngalert/api/tooling/definitions/alertmanager_test.go:403
            	Error:      	Expected nil, but got: &yaml.TypeError{Errors:[]string{"line 11: cannot unmarshal !!map into json.RawMessage"}}
            	Test:       	Test_PostableApiReceiver_Unmarshaling_YAML/grafana_receivers
FAIL
```

**Review by gotjosh — APPROVED** [review]
LGTM

**Pull request merged**